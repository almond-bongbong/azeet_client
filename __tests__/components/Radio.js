import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Radio from 'components/Radio/Radio';

describe('Radio', () => {
  it('matches snapshot', () => {
    const handleChange = jest.fn();
    const utils = render(<Radio onChange={handleChange} id="radio_0" />);

    expect(utils.container).toMatchSnapshot();
  });

  it('onChange radio', () => {
    const handleChange = jest.fn();
    const { getByLabelText } = render(
      <>
        <Radio onChange={handleChange} id="radio_0" label="male" checked />
        <Radio onChange={handleChange} id="radio_1" label="female" />
      </>,
    );

    getByLabelText('male');
    const radioForFemale = getByLabelText('female');

    fireEvent.click(radioForFemale);

    expect(handleChange).toBeCalled();
  });
});
