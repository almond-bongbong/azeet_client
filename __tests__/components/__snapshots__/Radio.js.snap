// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio matches snapshot 1`] = `
<div>
  <div
    class="Radio__BasicRadioStyle-sc-1ewdkdh-0 bQpYmY"
    label=""
  >
    <label
      for="radio_0"
    >
      <input
        id="radio_0"
        type="radio"
        value=""
      />
      <span />
    </label>
  </div>
</div>
`;
